auraed/src/runtime/cells/cells.rs

```diff
@@ -84,17 +92,63 @@ impl CellsTable {
         }
     }
 
-    /// Remove and return the cgroup keyed by [cell_name] from the cache.
-    /// Returns an error if the cell_name does not exist in the cache.
-    pub async fn remove(&self, cell_name: &CellName) -> Result<Cell> {
+    pub async fn get_mut<F, R>(&self, cell_name: &CellName, f: F) -> Result<R>
+    where
+        F: FnOnce(&mut Cell) -> Result<R>,
+    {
         let mut cache = self.cache.lock().await;
+        get_mut(&mut cache, cell_name, f)
+    }
 
+    /// Remove and return the [Cell] keyed by [CellName] from [Cells].
+    /// Returns an error if the [CellName] does not exist in the cache.
+    pub async fn free(&self, cell_name: &CellName) -> Result<Cell> {
```

Comment:
it's a tiny bit odd to me that something called "free" returns a valid item.  i would expect that after "free" any Cell referenced is no longer valid to use (a hangover from C++ memory models i suppose).

perhaps this should be "remove" even though it's called by "free".

Reply:
Another option is to just return 'Result<()>` as nothing is using the result as of now.

Reply:
Changed it to return `Result<()>`